### PR TITLE
Added the ability to edit filename scraping rules [#801]

### PR DESCRIPTION
**Pull request by mcpierce**
* Renamed the table and APIs to clarify that it's filename scraping.
 * Added a new tab to the config page for filename scraping rules.

## Status
READY

## Does this PR contain migrations?
YES

## Before You Submit Your PR:
- [ ] Have you announced your PR on the comixed-dev mailing list?
- [X] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Refactors existing code (code changes for efficiency or maintainability)
- [ ] Security fix (be sure to include the CVE in the commit message) 
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [X] I have added tests to cover my changes.
- [X] All new and existing tests passed.

### CODE REVIEW

File: comixed-adaptors/src/test/java/org/comixedproject/adaptors/comicbooks/FilenameScraperAdaptorTest.java [hunk 1]
```diff
@@ -53,29 +54,30 @@
           "{0} Vol.{1} #{2} (Farkle, 2019).cbz", TEST_SERIES, TEST_VOLUME, TEST_ISSUE_NUMBER);
 
   @InjectMocks private FilenameScraperAdaptor adaptor;
-  @Mock private ScrapingRule scrapingRule;
+  @Mock private Comic comic;
```
Comment (BRUCELLA2) [inline comixed-adaptors/src/test/java/org/comixedproject/adaptors/comicbooks/FilenameScraperAdaptorTest.java]: Comic is never used here.

File: comixed-services/src/main/java/org/comixedproject/service/scraping/FilenameScrapingRuleService.java [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+    final List<FilenameScrapingRule> rules = this.loadRules();
+    for (int index = 0; index < rules.size(); index++) {
+      final FilenameScrapingRule rule = rules.get(index);
+      log.trace("Testing filename scraping rule: priority={}", rule.getPriority());
+      final FilenameMetadata result = this.filenameScraperAdaptor.execute(filename, rule);
+      if (result.isFound()) {
+        log.trace("Rule applied!");
+        return result;
+      }
+    }
```
Comment (BRUCELLA2) [inline comixed-services/src/main/java/org/comixedproject/service/scraping/FilenameScrapingRuleService.java]: Need to update the for loop to enhanced for loop :
for (final FilenameScrapingRule rule : rules) {
      log.trace("Testing filename scraping rule: priority={}", rule.getPriority());
      final FilenameMetadata result = this.filenameScraperAdaptor.execute(filename, rule);
      if (result.isFound()) {
        log.trace("Rule applied!");
        return result;
      }
    }

**Review by BRUCELLA2 — CHANGES_REQUESTED** [review]
Just 2 comments otherwise it's ok for me.

File: comixed-adaptors/src/test/java/org/comixedproject/adaptors/comicbooks/FilenameScraperAdaptorTest.java (hunk 1 above)
Comment (mcpierce) [reply comixed-adaptors/src/test/java/org/comixedproject/adaptors/comicbooks/FilenameScraperAdaptorTest.java]: Yeah, hit some weird merging issues when I rebased this on top of #46. I'll clean this up today and push and update.

File: comixed-services/src/main/java/org/comixedproject/service/scraping/FilenameScrapingRuleService.java (hunk 2 above)
Comment (mcpierce) [reply comixed-services/src/main/java/org/comixedproject/service/scraping/FilenameScrapingRuleService.java]: Will do.

**Comment by mcpierce** [comment]
@BRUCELLA2 I also renamed the migration to avoid a collision with #866.

**Review by BRUCELLA2 — APPROVED** [review]

**Pull request merged**